desc/experimental/_terpsichore.py

```diff
@@ -0,0 +1,872 @@
+from desc.transform import Transform
+from desc.utils import errorif, warnif
+from desc.vmec_utils import ptolemy_identity_rev, zernike_to_fourier
+
+
+@execute_on_cpu
+def terpsichore(
+    eq,
+    processes=1,
+    path="",
+    exec="",
+    scalar=True,
+    mode_family=-1,
+    surfs=101,
+    lssl=1000,
+    lssd=1000,
+    M_max=8,
+    N_min=-4,
+    N_max=4,
+    M_booz_max=19,
+    N_booz_max=18,
+    awall=2.0,
+    deltaJp=1e-2,
+    modelk=1,
+    nev=1,
+    al0=-5e-1,
+    sleep_time=1,
+    stop_time=60,
+    data_transforms=None,
+    fit_transform=None,
+    theta0_outboard=True,
+):
+    """TERPSICHORE driver function."""
+    # TERPSICHORE assumes theta=0 is on the outboard midplane
+    if not theta0_outboard:
+        eq = flip_theta(eq)
+
+    # create temporary directory to store I/O files
+    tmp_path = os.path.join(path, "tmp-TERPS")
+    os.mkdir(tmp_path)
+
+    # write input files for each equilibrium in serial
+    idxs = list(range(len(eq)))  # equilibrium indices
+    for k in idxs:
+        idx_path = os.path.join(tmp_path, str(k))
+        os.mkdir(idx_path)
+        exec_path = os.path.join(idx_path, exec)
+        input_path = os.path.join(idx_path, "input")
+        wout_path = os.path.join(idx_path, "wout.nc")
+        shutil.copy(os.path.join(path, exec), exec_path)
+        _write_wout(
+            eq=eq[k],
+            path=wout_path,
+            surfs=surfs,
+            data_transforms=data_transforms,
+            fit_transform=fit_transform,
+        )
+        _write_terps_input(
+            path=input_path,
+            mode_family=mode_family,
+            surfs=surfs,
+            lssl=lssl,
+            lssd=lssd,
+            M_max=M_max,
+            N_min=N_min,
+            N_max=N_max,
+            M_booz_max=M_booz_max,
+            N_booz_max=N_booz_max,
+            awall=awall,
+            deltaJp=deltaJp,
+            modelk=modelk,
+            nfp=eq[k].NFP,
+            nev=nev,
+            al0=al0,
+        )
+
+    # run TERPSICHORE on list of equilibria in parallel
+    if len(eq) == 1:  # no multiprocessing if only one equilibrium
+        result = jnp.atleast_1d(
+            _pool_fun(
+                0,
+                path=tmp_path,
+                exec=exec,
+                scalar=scalar,
+                surfs=surfs,
+                al0=al0,
+                sleep_time=sleep_time,
+                stop_time=stop_time,
+            )
+        )
+    else:
+        with mp.Pool(processes=min(processes, len(eq))) as pool:
+            results = pool.map(
+                functools.partial(
+                    _pool_fun,
+                    path=tmp_path,
+                    exec=exec,
+                    scalar=scalar,
+                    surfs=surfs,
+                    al0=al0,
+                    sleep_time=sleep_time,
+                    stop_time=stop_time,
+                ),
+                idxs,
+            )
+            pool.close()
```

Comment:
I am pretty sure the `with` statement will call `pool.terminate` which is `pool.close` + `pool.join` when it exits the block, so you could probably remove these last three lines in the block and instead just have the last `result=` line be outside of the `with` block